### PR TITLE
[JUJU-1039] Rewrite `controller-config` to use `ConfigCommandBase`

### PR DESCRIPTION
**Pull request by barrettj12**
Continues work done in #13983 and #14014. I had to make a small retroactive change to ConfigCommandBase (allowing child commands to disable --reset flag), since the controller config API doesn't support resetting values. **Reviewers:** it will make more sense looking at the commits individually.

Again, there will be slight behaviour changes to the `controller-config` CLI:
- Config yaml files must now be specified using the --file flag.
- We no longer allow multiple actions to be done simultaneously (e.g. setting from key=value args & setting from file), as this creates ambiguity as to which should be done first. As a result, I removed several unit tests in `configcommand_test.go` which didn't make sense anymore.

### CODE REVIEW

**Review by benhoyt — APPROVED** [review]
Looks good to me, just a few small comments.

File: cmd/juju/controller/configcommand.go [hunk 1]
```diff
@@ -7,50 +7,60 @@ import (
...
 	apicontroller "github.com/juju/juju/api/controller/controller"
 	jujucmd "github.com/juju/juju/cmd"
 	"github.com/juju/juju/cmd/juju/common"
+	"github.com/juju/juju/cmd/juju/config"
 	"github.com/juju/juju/cmd/modelcmd"
 	"github.com/juju/juju/cmd/output"
 	"github.com/juju/juju/controller"
 )
 
+var ctrConfigBase = config.ConfigCommandBase{
```
[the remark this answers is not in the document]
Comment (barrettj12) [reply cmd/juju/controller/configcommand.go:30]: Whoops, should be used in `NewConfigCommandForTest` too

File: cmd/juju/controller/configcommand.go [hunk 2]
```diff
@@ -7,50 +7,60 @@ import (
...
+	Resettable: false,
+}
+
 // NewConfigCommand returns a new command that can retrieve or update
 // controller configuration.
 func NewConfigCommand() cmd.Command {
-	return modelcmd.WrapController(&configCommand{})
+	return modelcmd.WrapController(&configCommand{configBase: ctrConfigBase})
```
[the remark this answers is not in the document]
Comment (barrettj12) [reply cmd/juju/controller/configcommand.go:37]: That's true, I feel like it's better to be explicit here tho

File: cmd/juju/controller/configcommand.go [hunk 3]
```diff
@@ -202,41 +195,73 @@ func (c *configCommand) getConfig(client controllerAPI, ctx *cmd.Context) error
...
+
+// setConfigFile sets config values from the provided yaml file.
+func (c *configCommand) setConfigFile(client controllerAPI, ctx *cmd.Context) error {
+	// Read file & unmarshal into yaml
+	attrs := make(map[string]interface{})
+	path, err := utils.NormalizePath(c.configBase.ConfigFile.Path)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	data, err := ioutil.ReadFile(ctx.AbsPath(path))
```
[the remark this answers is not in the document]
Comment (barrettj12) [reply cmd/juju/controller/configcommand.go:233]: Ok didn't realise, literally just copy-pasted from `cmd/juju/common/flags.go:ReadAttrs`

**Comment by barrettj12** [comment]
$$merge$$

**Pull request merged**